cpp/unit/unit_texture.cpp

```diff
@@ -145,7 +145,7 @@ unsigned int dir_group(coord::phys3_delta dir, unsigned int angles) {
 	unsigned int first_angle = 5 * angles / 8;
 
 	// normalise direction vector
-	double len = std::hypot(dir.ne, dir.se);
+	double len = std::hypot(dir.ne, dir.se).to_double();
```

Comment:
Users will indeed expect `operator /(A a, B b)` to return `A`, you're right.

While `a * 0.5` should work as expected, I think we'd best not support `0.5 * a`.

__However__, in this particular use case, we don't want that behavior of the division operator. We know that the result will be in the interval `[-1; 1]`, and there `phys_t` gives us only 16 bits, significantly less than `float` or something like`FixedPoint<int64_t, 32>`. It's not that bad for `phys_t`, but other types such as `tile_t` have lower resolution (0 bits).
So I suggest adding a templated member function, `div`, where the caller can specify their own return type (`float`, `FixedPoint<...>`, ...).